tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/ContinuousIntegrationPipelineConvention.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.Build.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PipelineGenerator.Conventions
+{
+    public abstract class ContinuousIntegrationPipelineConvention : PipelineConvention
+    {
+        private const string ReportBuildStatusKey = "reportBuildStatus";
+
+        public ContinuousIntegrationPipelineConvention(ILogger logger, PipelineGenerationContext context) : base(logger, context)
+        {
+        }
+
+        protected override Task<bool> ApplyConventionAsync(BuildDefinition definition, SdkComponent component)
+        {
+            // NOTE: Not happy with this code at all, I'm going to look for a reasonable
+            // API that can do equality comparisons (without having to do all the checks myself).
+
+            var hasChanges = false;
+
+            if (EnsureVariableGroups(definition))
+            {
+                hasChanges = true;
+            }
+
+            if (definition.Path != $"\\{this.Context.Prefix}")
+            {
+                definition.Path = $"\\{this.Context.Prefix}";
+                hasChanges = true;
+            }
+
+            var ciTrigger = definition.Triggers.OfType<ContinuousIntegrationTrigger>().SingleOrDefault();
```

Comment:
OK - I probably need to be less specific on the filter here then so that we just override whatever is there.

Reply:
Looking at this, I don't think this would be the case. If there is no trigger it would add one, if there is already a trigger it would switch it to source settings type 2. Am I missing something?